libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -1361,6 +1361,18 @@ bool ExpressionCompiler::visit(MemberAccess const& _memberAccess)
 			);
 			m_context << Instruction::POP;
 		}
+		else if (member == "name")
+		{
+			TypePointer arg = dynamic_cast<MagicType const&>(*_memberAccess.expression().annotation().type).typeArgument();
+			ContractDefinition const& contract = dynamic_cast<ContractType const&>(*arg).contractDefinition();
+			m_contract << u256(contract.name().length() + 32);
+			utils().allocateMemory();
```

Comment:
How about an overload where `allocateMemory` takes an `unsigned`?

I see in many cases the pattern in
```
m_context << number;
allocateMemory():
```

With the overload we could avoid the `SWAP` at this stage.

Reply:
Sure!

Reply:
Created an issue #6106